### PR TITLE
fix: Azure Pipelines Scaler uses correct endpoint when demands are set  (#4387)

### PR DESCRIPTION
**Pull request by Eldarrin**
Added a simple hotfix to grab different querystring excluding top when parent is required,

### Checklist

- [x] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #
#4387 

Relates to #

### CODE REVIEW

**Review by JorTurFer — APPROVED** [review]

File: tests/scalers/azure/azure_pipelines_adv/azure_pipelines_adv_test.go [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+	"k8s.io/client-go/kubernetes"
+
+	. "github.com/kedacore/keda/v2/tests/helper"
+)
+
+// Load environment variables from .env file
+var _ = godotenv.Load("../../../.env")
+
+const (
+	testName = "azure-pipelines-test"
```
Comment (JorTurFer) [inline tests/scalers/azure/azure_pipelines_adv/azure_pipelines_adv_test.go:31]: Use other namespace than the already e2e test for avoid conflicts 
```suggestion
	testName = "azure-pipelines-demands-test"
```

**Comment by Eldarrin** [comment]
@JorTurFer  I think running the new tests in parallel with the old tests upset ADO :)

**Comment by Eldarrin** [comment]
I can mitigate it, does your copy of ADO support running parallel self hosted jobs? The basic ado account doesn’t

**Comment by JorTurFer** [comment]
> I can mitigate it, does your copy of ADO support running parallel self hosted jobs? The basic ado account doesn’t

I think so:
image

**Comment by Eldarrin** [comment]
Ok, will do in the morning and update

**Comment by JorTurFer** [comment]
> Ok, will do in the morning and update

Sure, no rush! I have just checked GH after a disconnecting weekend 😄

**Comment by JorTurFer** [comment]
> Ok, so new patch should allow e2e to run in parallel, can't really test at home as only have a basic account for ADO.
You will need to add a new agent pool (can be empty), change the new pipeline to point to the new pool and put the poolname as the new .env variable added for AZURE_DEVOPS_DEMAND_POOL_NAME=

@tomkerkhove, Could you do it? I don't have enough permissions in AzDO org

**Comment by JorTurFer** [comment]
I guess that we need to update also the pipeline definition to use the new agent, right?

**Review by JorTurFer — CHANGES_REQUESTED** [review]
Could you add a record in the changelog? I didn't notice it before

**Comment by Eldarrin** [comment]
> Could you add a record in the changelog? I didn't notice it before

Will do in the morning :)

**Review by JorTurFer — APPROVED** [review]

**Pull request merged**